### DISCUSSION TITLE
Allow Once vs. Allow While Using App

### DISCUSSION THREAD

**Discussion by jorjimen** [Agontuk/react-native-geolocation-service#380] [Q&A]
When calling `Geolocation.requestAuthorization`, the iOS modal has the options between Allow Once and Allow While Using App. If the user clicks either of these options, the return type is always `granted` 

Is there any way that we can differentiate the `granted` response depending on the option that the user selected? I wanted to take a different action depending on what the user has selected.

**Comment by Agontuk** [comment]
I recommend you to use `react-native-permissions` library for permission related feature. `requestAuthorization` will be removed in future releases.
